tests/ui/lifetimes/issue-90600-expected-return-static-indirect.stderr

```diff
@@ -7,7 +7,7 @@ LL |     let refcell = RefCell::new(&mut foo);
    |                                ^^^^^^^^ borrowed value does not live long enough
 LL |
 LL |     let read = &refcell as &RefCell<dyn Read>;
-   |                -------- cast requires that `foo` is borrowed for `'static`
+   |                -------- coercion requires that `foo` is borrowed for `'static`
```

Comment:
this and some others bellow sure look like a cast?

Reply:
This is a side effect from how explicit coercion casts and implicit coercions are represented in the same way in MIR. I will try to recover the wording here.

Reply:
So this turned out to be non-trivial, because there is currently no way to distinguish
```rust
let x: T = y;
```
from
```rust
let x = y as T
```
by looking at the MIR alone (without relying on the exact order of MIR statements).

So our options here are
* Leave the diagnostic as is, calling some coercion casts just "coercions" (technically correct).
* Revert the wording and call all coercions "casts" (technically wrong).
* Add some additional diagnostic-only information into the MIR to allow us to distinguish them and output the optimal diagnostic (which is really just one word). This is currently unprecedented as far as I'm aware.

I'm unsure what the best option here is and would appreciate your input :)

Reply:
I would prefer the third option, i.e. to add some diagnostic only info. That being said I think you should check with people who work on MIR more (miri or miropt zulip streams maybe?).

The first option (using "coercion" when the user wrote a cast) could be OK. If you want we can leave the third option as a follow up (please try to do it though 😄).

Reply:
After checking again, I found that we do have one instance of diagnostic information in MIR here, so this feels less unprecedented now: https://github.com/rust-lang/rust/blob/8c2c9a9ef527edb453bcee638d34beb0e7b4616b/compiler/rustc_middle/src/mir/syntax.rs#L735-L736